tt_metal/impl/allocator/l1_banking_allocator.cpp

```diff
@@ -117,12 +117,12 @@ void init_compute_and_storage_l1_bank_manager(Allocator &allocator, const Alloca
         num_banks.total
     );
 
-    uint32_t reserved_region = (alloc_config.l1_bank_size == alloc_config.worker_l1_size) ? L1_UNRESERVED_BASE : STORAGE_ONLY_UNRESERVED_BASE;
     // There is only alloc_config.l1_bank_size bytes available for L1 buffers to be allocated in
-    uint64_t allocatable_l1_size = static_cast<uint64_t>(alloc_config.l1_bank_size) - reserved_region;
+    uint64_t interleaved_address_limit = static_cast<uint64_t>(alloc_config.worker_l1_size - alloc_config.l1_bank_size) + STORAGE_ONLY_UNRESERVED_BASE;
+    uint64_t allocatable_l1_size = static_cast<uint64_t>(alloc_config.worker_l1_size) - L1_UNRESERVED_BASE;
     // Assuming top down allocation for L1 buffers so the allocatable memory space is the top alloc_config.l1_bank_size bytes of L1
-    uint64_t alloc_offset = static_cast<uint64_t>(alloc_config.worker_l1_size - alloc_config.l1_bank_size) + reserved_region;
-    allocator.l1_manager = BankManager(BufferType::L1, bank_id_to_bank_offset, allocatable_l1_size, alloc_offset);
+    uint64_t alloc_offset = L1_UNRESERVED_BASE;
+    allocator.l1_manager = BankManager(BufferType::L1, bank_id_to_bank_offset, allocatable_l1_size, interleaved_address_limit, alloc_offset);
```

Comment:
because the allocator changes are included could you be sure to test this on all the different device configs we have. These are specified in the soc desc yamls for each arch
Grayskull: E150 and E75
Wormhole: Nebula 1x, Nebula 2x, Galaxy (no harvested rows)

Reply:
@tt-nshanker could you look into this regarding the allocater

Reply:
@abhullar-tt where are the current allocater tests? and how do I run them ? is this a pipeline or does this involve writing more tests?